File: src/components/info/augment/gigas.ts
import Augment, { AugmentGroup } from "./_base";
import { createMultipleAugments } from "./_util";
import { EFFECTS as eff, SINGLE_OFF_EFF } from "../_effect";

const GROUP = "GIGAS";

let augments: Augment[] = [];

["might", "precision", "technique"].forEach((name, index) => {
    createMultipleAugments(
        `gigas ${name}`,
        3,
        [
            { effect: eff.HP, amount: [5, 10, 15] },
            ...SINGLE_OFF_EFF[index].map((effect) => {
                return { effect, amount: [1.015, 1.02, 1.025] };
            }),
        ],
        [6, 8, 10],
        GROUP,
    ).forEach((aug) => {
        augments.push(aug);
    });
});

const GIGAS: AugmentGroup = {
    name: GROUP,
    augments,
};

export default GIGAS;

File: _data/augments/gigas.py
from typing import List

from ._augment import Augment
from ._augment_groups import AugmentGroups
from effect import *
from effect import EffectTypes as ET


GROUP = AugmentGroups.GIGAS
CONFLICT = (GROUP,)

augments: List[Augment] = []

# ----------------------------------------------------------------
_names = ("might", "precision", "technique")
for name, eff in zip(_names, OFFENSIVE_POT):
    augments.extend(
        Augment.from_list(
            f"gigas {name}",
            3,
            (6, 8, 10),
            (
                EffectWithManyAmount(ET.HP, (10, 15, 30)),
                EffectWithManyAmount(eff, (1.015, 1.02, 1.025)),
            ),
            GROUP,
            CONFLICT,
        )
    )
# ----------------------------------------------------------------
